src/Text/Pandoc/Writers/Txt2Tags.hs

```diff
@@ -0,0 +1,512 @@
+{-# LANGUAGE OverloadedStrings #-}
+{- |
+   Module      : Text.Pandoc.Writers.Txt2Tags
+   Copyright   : Copyright (C) 2020 Eric Forgeot, based on John MacFarlane dokuwiki writer
+   License     : GNU GPL, version 2 or above
+
+   Maintainer  : Eric Forgeot - https://github.com/farvardin/
+   Stability   : alpha
+   Portability : portable
+
+Conversion of 'Pandoc' documents to Txt2Tags markup.
+
+Txt2Tags:  <https://www.Txt2Tags.org/>
+-}
+
+{-
+   It works for most of the syntax. But some improvements or fix can be made:
+   - On lists (2 extra lines to terminate a list)
+   - On the 3 lines header at the begining of a file
+   - Definition lists are broken
+   - Tables could be improved
+   - Some formats make better results than others (html is ok, md is somehow broken)
+   - code related to dokuwiki only could be removed
+   - a test case should be made
+-}
+
+module Text.Pandoc.Writers.Txt2Tags ( writeTxt2Tags ) where
+import Control.Monad (zipWithM)
+import Control.Monad.Reader (ReaderT, asks, local, runReaderT)
+import Control.Monad.State.Strict (StateT, evalStateT)
+import Data.Default (Default (..))
+import Data.List (intersect, transpose)
+import Data.Text (Text)
+import qualified Data.Text as T
+import Text.Pandoc.Class.PandocMonad (PandocMonad, report)
+import Text.Pandoc.Definition
+import Text.Pandoc.ImageSize
+import Text.Pandoc.Logging
+import Text.Pandoc.Options (WrapOption (..), WriterOptions (writerTableOfContents, writerTemplate, writerWrapText))
+import Text.Pandoc.Shared (camelCaseToHyphenated, escapeURI, isURI, linesToPara,
+                           removeFormatting, trimr, tshow)
+import Text.Pandoc.Templates (renderTemplate)
+import Text.DocLayout (render, literal)
+import Text.Pandoc.Writers.Shared (defField, metaToContext, toLegacyTable)
+
+data WriterState = WriterState {
+  }
+
+data WriterEnvironment = WriterEnvironment {
+    stIndent      :: Text          -- Indent after the marker at the beginning of list items
+  , stUseTags     :: Bool            -- True if we should use HTML tags because we're in a complex list
+  , stBackSlashLB :: Bool     -- True if we should produce formatted strings with newlines (as in a table cell)
+  }
+
+instance Default WriterState where
+  def = WriterState {}
+
+instance Default WriterEnvironment where
+  def = WriterEnvironment { stIndent = ""
+                          , stUseTags = False
+                          , stBackSlashLB = False }
+
+type Txt2Tags m = ReaderT WriterEnvironment (StateT WriterState m)
```

Comment:
Looks like we don't need the `WriterState`, so this could shorten this to `ReaderT WriterEnvironment m`. The function `runTxt2Tags` would then become `runTxt2Tags = flip runReaderT def`.

Reply:
I don't know anything about haskell, so I don't understand the difference, sorry :(  

![image](https://user-images.githubusercontent.com/477243/104966264-bc33f480-59e0-11eb-8318-d67c1662a5b7.png)

Reply:
This is copy-pasted together without you knowing Haskell?

![I'm not even mad. That's amazing!](https://user-images.githubusercontent.com/507994/105008481-9555dc80-5a39-11eb-9ff5-ed79cf291142.png)

Reply:
> I think the main stumbling block for us will be maintenance. We have to ensure that this will be maintained after it's merged. Not sure how to proceed there, I'm not too keen to up to take up an additional responsibility. @jgm, what do you think?

the txt2tags language is not evolving, for 2 reasons: it's stable for many years, and also it's possible to add custom markup with some preproc and postprocessors, so the core doesn't need changing. So the only maintenance would be to adapt the writer code in the case pandoc change something it it's internal code. But it would require the same maintenance as the dokuwiki code.